### ISSUE TITLE
Non random data

### ISSUE THREAD

**Issue by deibraz-free** [pietz/language-recognition#10]
Hey, so I am trying to make the system work with very specific data (non randomised train,valid,test data), No matter what, the accuracy never goes above 60%. With your randomised code, same dataset, it goes up to 90% (in accuracy measurement), but I need to add specific datasets for train/valid/test.

Any chance you could help somehow?

Newly created list looks something like this: https://gyazo.com/7c6222c37cc8176f3af541ef310709d6 .

Other thing I've noticed was that the main data file y IDs are different, so instead of "dask.array.image.imread('data/jpg/*.jpg')" I do each file individually, combine it. Still gives me 60% only, accuracy. This code seems to work, outputs of *.jpg and the code bellow match (with 2 layers ordered as needed)
```
# Manually load all images into one dataset
def loadAllImages(filePath = ""):
    removeFile(filePath)

    d_list = pd.read_csv("data/list_full.csv")

    d = np.zeros(shape=(18000, 192, 192), dtype="uint8")
    for i, j in d_list.iterrows():
        di = dask.array.image.imread('data/jpg/' + j["filename"] + '.jpg')
        dn = di.compute()
        d[i] = dn

    ddask = da.from_array(d, chunks=(1, 192, 192))

    ddask.to_hdf5(filePath, 'data')
    print("Finished compiling data", ddask, ddask.shape, ddask.size, type(ddask))

```

And instead of the random picking of train/valid/test I've changed to this code, which should in theory represent it perfectly. This matches the .csv list of files

```
    full_idx = []
    for i in range(data_size*out_dim):
    full_idx.append(i)

    full_idx = np.array(full_idx)

    tr_idx = full_idx[:(tr_size*out_dim)]
    va_idx = full_idx[(tr_size*out_dim):(tr_size*out_dim) + (va_size*out_dim)]
    te_idx = full_idx[(tr_size*out_dim) + (va_size*out_dim):]
```

I would love you forever if I could get any more of your help on this one, as this is an amazing system that I would wish to get working propely.

**Comment by pietz** [comment]
You're working with a different dataset, correct? There is little I can do with the information you provide and I'm sorry to say that I don't have more time to help debugging. However, from what you tell me I can give you a few pointers:

- You should always create the "best possible dumb baseline" you can come up with. If yours is a binary problem and one class makes up 60% of the samples, that's where the 60% may come from. Could this be the case?
- Look at the predictions in details. The confusion matrix is often a good methods that doesn't require much work. Does your network predict only 1 class? Is there a pattern in the confusion matrix? The answer to both should be "No".
- 90% acc on train and 60% acc on validation usually (!) means one of 4 things: 1) The model cannot find a correlation and will simply overfit on the training data. 2) There is a correlation but your dataset is so small (model is so large) that you're overfitting instantly. 3) You messed up when creating the train/validation split 4) The model only predicts one value (usually 0)

Check those things please.

**Comment by deibraz-free** [comment]
Been trying to sort this issue out.

- 5 languages with ideally same number of samples, same sample size

- It predicts several, but failing quite a bit randomly

- Thing is, I have 3600 sound files (10s each), but with only around 200 speakers. train/valid/test sets don't have same speakers in  multiple sets. If I try to shuffle the data manually and only use "validation_split" in model fitting, val_acc and acc seem to be a lot closer. Any ideas?

**Comment by pietz** [comment]
The last point makes it sound like the CNN is simply learning the 200 voices, which would also explain the low validation accuracy when you properly handle the leakage of speakers. Anyway, there is still little I can help you with nor does it seem to be an issue with the code. I'm closing this issue.

Best of luck. Maybe you can try stackoverflow with your questions.
